guest-agent/VfioService/VfioService/InterceptMouse.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Drawing;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace VfioService
+{
+    public class InterceptMouse
+    {
+        private const int WH_MOUSE_LL = 14;
+        private const int WM_MOUSEMOVE = 0x0200;
+
+        private static LowLevelMouseProc _proc = HookCallback;
+        private static IntPtr _hookID = IntPtr.Zero;
+        public static ClientManager ClientManager { get; set; }
+
+        public static void Start()
+        {
+            SetProcessDpiAwareness(ProcessDPIAwareness.ProcessPerMonitorDPIAware);
+            _hookID = SetHook(_proc);
+        }
+        public static void Stop()
+        {
+            UnhookWindowsHookEx(_hookID);
+        }
+
+        private static IntPtr SetHook(LowLevelMouseProc proc)
+        {
+            using (Process curProcess = Process.GetCurrentProcess())
+            using (ProcessModule curModule = curProcess.MainModule)
+            {
+                return SetWindowsHookEx(WH_MOUSE_LL, proc, GetModuleHandle(curModule.ModuleName), 0);
+            }
+        }
+
+        private delegate IntPtr LowLevelMouseProc(int nCode, IntPtr wParam, IntPtr lParam);
+
+        private static IntPtr HookCallback(int nCode, IntPtr wParam, IntPtr lParam)
+        {
+            if (nCode >= 0 && wParam.ToInt32() == WM_MOUSEMOVE)
+            {
+                MSLLHOOKSTRUCT hookStruct = (MSLLHOOKSTRUCT)Marshal.PtrToStructure(lParam, typeof(MSLLHOOKSTRUCT));
+                var inside = false;
+                foreach (var s in Screen.AllScreens)
+                {
+                    inside |= s.Bounds.Contains(new Point(hookStruct.pt.x, hookStruct.pt.y));
```

Comment:
This needs a comment explaining why it works. Shouldn't the cursor always be inside a screen? Presumably this fails specifically on edges but either way: Document it!